File: java/junit4-kickstart-project/src/main/java/com/codurance/katalyst/kickstart/service/UserService.java
package com.codurance.katalyst.kickstart.service;

import com.codurance.katalyst.kickstart.model.User;
import com.codurance.katalyst.kickstart.repository.UserRepository;

public class UserService {

    private UserRepository userRepository;

    public int createNewUser(User newUser) throws DuplicateUserException {
        if (userRepository.getUser(newUser.getUserName()) != null) {
            throw new DuplicateUserException("User with name " + newUser.getUserName() + " is already registered");
        }

        return userRepository.add(newUser);
    }

    public void setUserRepository(UserRepository userRepository) {
        this.userRepository = userRepository;
    }
}

File: Calculator/Service/UserService.cs
using Calculator.Model;
using Calculator.Repository;

namespace Calculator.Service
{
    public class UserService
    {
        private IUserRepository userRepository;

        public int CreateNewUser(User newUser)
        {
            if (userRepository.GetUser(newUser.GetUserName()) != null)
            {
                throw new DuplicateUserException("User with name " + newUser.GetUserName() + " is already registered");
            }

            return userRepository.Add(newUser);
        }

        public void SetUserRepository(IUserRepository userRepository)
        {
            this.userRepository = userRepository;
        }
    }
}